### ISSUE TITLE
The plugin "vagrant-bindfs" could not be found.

### ISSUE THREAD

**Issue by bchenu** [gael-ian/vagrant-bindfs#21]
I have an issue with vagrant 1.4.3. After installing the plugin, vagrant tells me that it can't find the plugin... I'm on ubuntu 14.10, here's the details :

$ vagrant plugin install vagrant-bindfs
Installing the 'vagrant-bindfs' plugin. This can take a few minutes...
Installed the plugin 'vagrant-bindfs (0.3.2)'!

$ vagrant up
The plugin "vagrant-bindfs" could not be found. Please make sure that it is
properly installed via `vagrant plugin`. Note that plugins made for
Vagrant 1.0.x are not compatible with 1.1+ and this error will likely
continue to show when you use `plugin install` with a 1.0.x plugin.

Bringing machine 'default' up with 'virtualbox' provider...
There are errors in the configuration of this machine. Please fix
the following errors and try again:

Vagrant:
* Unknown configuration section 'bindfs'.

$ vagrant -v
Vagrant 1.4.3

**Comment by gael-ian** [comment]
Hi,

It seems strange as vagrant-bindfs is compatible with the 1.1+ plugin API for more than a year and was even specially fixed for vagrant 1.4 in December 2013. All should be right.

Can you post your Vagrantfile ?

**Comment by bchenu** [comment]
Here's m'y Vagrant file :

# -*- mode: ruby -*-
# vi: set ft=ruby :

# Vagrantfile API/syntax version. Don't touch unless you know what you're doing!
VAGRANTFILE_API_VERSION = "2"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  # All Vagrant configuration is done here. The most common configuration
  # options are documented and commented below. For a complete reference,
  # please see the online documentation at vagrantup.com.

  # Every Vagrant virtual environment requires a box to build off of.
  config.vm.box = "CentOS-6.6-64"
  
  # Disable automatic box update checking. If you disable this, then
  # boxes will only be checked for updates when the user runs
  # `vagrant box outdated`. This is not recommended.
  # config.vm.box_check_update = false

  # Create a forwarded port mapping which allows access to a specific port
  # within the machine from a port on the host machine. In the example below,
  # accessing "localhost:8080" will access port 80 on the guest machine.
  # config.vm.network "forwarded_port", guest: 80, host: 8080

  # Create a private network, which allows host-only access to the machine
  # using a specific IP.
  # config.vm.network "private_network", ip: "192.168.33.10"
  
  config.vm.network "private_network", type: :dhcp  

  # Create a public network, which generally matched to bridged network.
  # Bridged networks make the machine appear as another physical device on
  # your network.
  # config.vm.network "public_network"

  # If true, then any SSH connections made will enable agent forwarding.
  # Default value: false
  # config.ssh.forward_agent = true

  # Share an additional folder to the guest VM. The first argument is
  # the path on the host to the actual folder. The second argument is
  # the path on the guest to mount the folder. And the optional third
  # argument is a set of non-required options.
  # config.vm.synced_folder "../data", "/vagrant_data"
  config.vm.synced_folder ".", "/vagrant-nfs", create: true, type: "nfs"
  config.bindfs.bind_folder "/vagrant-nfs/site", "/www/www.xxxxxxx.com/site", create: true, :owner => "40", :group => "40"

  # Provider-specific configuration so you can fine-tune various
  # backing providers for Vagrant. These expose provider-specific options.
  # Example for VirtualBox:
  #
  config.vm.provider "virtualbox" do |vb|
  #   # Don't boot with headless mode
  #   vb.gui = true
  #
  #   # Use VBoxManage to customize the VM. For example to change memory:
     vb.customize ["modifyvm", :id, "--memory", "1024"]
  end
  #
  # View the documentation for the provider you're using for more
  # information on available options.

  # Enable provisioning with CFEngine. CFEngine Community packages are
  # automatically installed. For example, configure the host as a
  # policy server and optionally a policy file to run:
  #
  # config.vm.provision "cfengine" do |cf|
  #   cf.am_policy_hub = true
  #   # cf.run_file = "motd.cf"
  # end
  #
  # You can also configure and bootstrap a client to an existing
  # policy server:
  #
  # config.vm.provision "cfengine" do |cf|
  #   cf.policy_server_address = "10.0.2.15"
  # end

  # Enable provisioning with Puppet stand alone.  Puppet manifests
  # are contained in a directory path relative to this Vagrantfile.
  # You will need to create the manifests directory and a manifest in
  # the file default.pp in the manifests_path directory.
  #
  # config.vm.provision "puppet" do |puppet|
  #   puppet.manifests_path = "manifests"
  #   puppet.manifest_file  = "site.pp"
  # end

  # Enable provisioning with chef solo, specifying a cookbooks path, roles
  # path, and data_bags path (all relative to this Vagrantfile), and adding
  # some recipes and/or roles.
  #
  # config.vm.provision "chef_solo" do |chef|
  #   chef.cookbooks_path = "../my-recipes/cookbooks"
  #   chef.roles_path = "../my-recipes/roles"
  #   chef.data_bags_path = "../my-recipes/data_bags"
  #   chef.add_recipe "mysql"
  #   chef.add_role "web"
  #
  #   # You may also specify custom JSON attributes:
  #   chef.json = { mysql_password: "foo" }
  # end

  # Enable provisioning with chef server, specifying the chef server URL,
  # and the path to the validation key (relative to this Vagrantfile).
  #
  # The Opscode Platform uses HTTPS. Substitute your organization for
  # ORGNAME in the URL and validation key.
  #
  # If you have your own Chef Server, use the appropriate URL, which may be
  # HTTP instead of HTTPS depending on your configuration. Also change the
  # validation key to validation.pem.
  #
  # config.vm.provision "chef_client" do |chef|
  #   chef.chef_server_url = "https://api.opscode.com/organizations/ORGNAME"
  #   chef.validation_key_path = "ORGNAME-validator.pem"
  # end
  #
  # If you're using the Opscode platform, your validator client is
  # ORGNAME-validator, replacing ORGNAME with your organization name.
  #
  # If you have your own Chef Server, the default validation client name is
  # chef-validator, unless you changed the configuration.
  #
  #   chef.validation_client_name = "ORGNAME-validator"
  
  config.vm.provision "ansible" do |ansible|
  	ansible.playbook = "provisioning/playbook.yml"
	#ansible.verbose = "vvv"
  end
end

**Comment by gael-ian** [comment]
Hi,

I've just tried a similar setup on my machine and everything works fine…

Can you give me the ouput of a `vagrant plugin list` ?

**Comment by bchenu** [comment]
$ vagrant plugin list
vagrant-bindfs (0.3.2)

This is the first time I install a plugin. I tried to check with an other plugin (vagrant-notify) and the problem is the same (same error). So this is not a vagrant-bindfs bug, this is a good point :)

All files a presents in ~/.vagrant.d/gems\:/usr/share/vagrant-plugins/gems/ directory and permissions are good. All seems to be fine (install, uninstall, list, ...) but Vagrant continue to send me this error.

I tried a vagrant up --debug but it does not provide a lot more informations : 

DEBUG environment: Loading plugins from: /home/bchenu/.vagrant.d/plugins.json
 INFO environment: Loading plugin from JSON: vagrant-bindfs
ERROR root: Failed to load plugin: vagrant-bindfs
ERROR root:  -- Error: #<LoadError: cannot load such file -- vagrant-bindfs>
ERROR root:  -- Backtrace:
ERROR root: /usr/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/usr/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/usr/lib/ruby/vendor_ruby/vagrant.rb:204:in `require_plugin'
/usr/lib/ruby/vendor_ruby/vagrant/environment.rb:760:in `block in load_plugins'
/usr/lib/ruby/vendor_ruby/vagrant/environment.rb:751:in `each'
/usr/lib/ruby/vendor_ruby/vagrant/environment.rb:751:in `load_plugins'
/usr/lib/ruby/vendor_ruby/vagrant/environment.rb:132:in `initialize'
/usr/bin/vagrant:105:in `new'
/usr/bin/vagrant:105:in `<main>'
 INFO interface: error: The plugin "vagrant-bindfs" could not be found. Please make sure that it is
properly installed via `vagrant plugin`. Note that plugins made for
Vagrant 1.0.x are not compatible with 1.1+ and this error will likely
continue to show when you use `plugin install` with a 1.0.x plugin.

The plugin "vagrant-bindfs" could not be found. Please make sure that it is
properly installed via `vagrant plugin`. Note that plugins made for
Vagrant 1.0.x are not compatible with 1.1+ and this error will likely
continue to show when you use `plugin install` with a 1.0.x plugin.


I don't know if you've already faced to this problem or have an idea... Otherwise, thanks for your time and sorry for this false issue :)

**Comment by gael-ian** [comment]
Hi,

Never faced anything similar, sorry. What is your host OS? (guess Linux but which flavor?)
Maybe you can try a fresh install of vagrant?

**Comment by bchenu** [comment]
This is a fresh install of ubuntu 14.10 with / a fresh install / of Vagrant. I will try with an other platform of a colleague.

**Comment by JulianVolodia** [comment]
Install `ruby-dev` using:
> sudo apt-get install ruby-dev
Next, try install `vagrant-bindfs` using `vagrant plugin install <name-of-plugin>` construction.

Is it helpful? I am downloading 14.10 now and I would check it for you. What version of virtualbox?

**Comment by bchenu** [comment]
I've installed ruby-dev and re-install vagrant-bindfs with vagrant plugin install command, same error :(

My virtualbox version : 
Oracle VM VirtualBox Manager 4.3.18_Ubuntu

**Comment by bchenu** [comment]
I tried on an other ubuntu 14.10 and reproduce de same problem. I tried on ubuntu 14.04 and don't have this error.

**Comment by bchenu** [comment]
Well, I finally found a post with the same problem :
https://github.com/opscode/vagrant-omnibus/issues/106

I've installed the latest version of vagrant (1.7.2) and this works.

I may found something : I've noted that plugin was installed in this directory :
~/vagrant.d/gems:/usr/share/vagrant-plugins/gems

And after installing the latest version, the plugin is installed in this one :
~/vagrant.d/gems/gems/

I've tried to rollback to vagrant 1.4.3 and tried to create a symlink to this directory and tried to copy without success in both case. I think there's an error in plugin path in this vagrant package on ubuntu 14.10, which explains that ruby can't find the file.
